### PR TITLE
BFD-2106: Update Synthea automation to handle creating and uploading future data

### PR DESCRIPTION
**Pull request by lsmitchell**
**JIRA Ticket:**
[BFD-2106](https://jira.cms.gov/browse/BFD-2106)

**User Story or Bug Summary:**

As a BFD Engineer I would like the Jenkins synthea automation job to support making loads with future data, so that we can have updating data to provide to users.

---

### What Does This PR Do?

Updates the Synthea automation to integrate the V2 synthea script, python script to split the future lines into their own loads, and finally upload the future loads to S3 if the new optional "generate future months" param is populated.

Changelog:
- Shell script
  - Updaded the "synthea_validate" flag in the shell script to be 'skip_validate' since that's what it was doing
  - Added new option for `--num_future_months` which if populated with a number > 0, will initiate logic to generate synthea lines that many months into the future
  - Removed S3 buckets from the script for security, per discussion with Burling
  - Passed new future months param into `prepare-and-run-synthea.py`
  - Added script execution for `split-future-claims.py` after the synthea generation if future months are set
  - Added logic to upload all the split future files to S3 if future months are set
- prepare-and-run-synthea.py
  - Added new param for num future months as param 5 (param 5 moved to param 6)
  - Added logic to call the V2 synthea script if future months > 0 and pass future months into it
- s3_utilities
  - Added slightly adjusted method for uploading the future folders to S3, since they are more complete and didn't need as much processing as the existing methods provided
- split-future-claims.py
  - Minor adjustment to check for file existence to help testing with smaller datasets where some files may not be generated, no functional change

### What Should Reviewers Watch For?

If you're reviewing this PR, please check for these things in particular:

* Verify all PR security questions and checklists have been completed and addressed.

### What Security Implications Does This PR Have?

Submitters should complete the following questionnaire:

* If the answer to any of the questions below is **Yes**, then **you must** supply a link to the associated Security Impact Assessment (SIA), security checklist, or other similar document in Confluence here: **N/A**

    * Does this PR add any new software dependencies? 
      * [ ] Yes
      * [x] No
    * Does this PR modify or invalidate any of our security controls?
      * [ ] Yes
      * [x] No
    * Does this PR store or transmit data that was not stored or transmitted before?
      * [ ] Yes
      * [x] No

* If the answer to any of the questions below is **Yes**, then please add @StewGoin as a reviewer, and note that this PR **should not be merged** unless/until he also approves it.
    * Do you think this PR requires additional review of its security implications for other reasons?
      * [ ] Yes
      * [x] No

### What Needs to Be Merged and Deployed Before this PR?

This PR cannot be either merged or deployed until the following prerequisite changes have been fully deployed:

* BFD-1912 must be merged

### Submitter Checklist

I have gone through and verified that...:

* [x] I have named this PR and branch so they are [automatically linked](https://confluence.atlassian.com/adminjiracloud/integrating-with-development-tools-776636216.html) to the (most) relevant Jira issue. Ie: `BFD-123: Adds foo`
* [x] This PR is reasonably limited in scope, to help ensure that:
    1. It doesn't unnecessarily tie a bunch of disparate features, fixes, refactorings, etc. together.
    2. There isn't too much of a burden on reviewers.
    3. Any problems it causes have a small "blast radius".
    4. It'll be easier to rollback if that becomes necessary.
* [x] This PR includes any required documentation changes, including `README` updates and changelog / release notes entries.
* [x] All new and modified code is appropriately commented, such that the what and why of its design would be reasonably clear to engineers, preferably ones unfamiliar with the project.
* [x] All tech debt and/or shortcomings introduced by this PR are detailed in `TODO` and/or `FIXME` comments, which include a JIRA ticket ID for any items that require urgent attention.
* [x] Reviews are requested from both:
    * At least two other engineers on this project, at least one of whom is a senior engineer or owns the relevant component(s) here.
    * Any relevant engineers on other projects (e.g. DC GEO, BB2, etc.).
* [x] Any deviations from the other policies in the [DASG Engineering Standards](https://github.com/CMSgov/cms-oeda-dasg/blob/master/policies/engineering_standards.md) are specifically called out in this PR, above.
    * Please review the standards every few months to ensure you're familiar with them.

### CODE REVIEW

**Review by mackec1 — COMMENTED** [review]
Looks reasonable; we'll gut much of the bash script once I get pipeline changes in.

**Review by mackec1 — DISMISSED** [review]
Looks good; depending on burling's Jenkins flow (pipeline) some of this may come out, but at least we have something to start with. thx.

File: ops/ccs-ops-misc/synthetic-data/scripts/synthea-automation/build_run_synthea.sh [hunk 1]
```diff
@@ -304,23 +321,55 @@ download_props_file_from_s3(){
...
+  readarray -t FOLDERS < <(find "${BFD_SYNTHEA_OUTPUT_LOCATION}" -maxdepth 1 -mindepth 1 -type d -execdir basename {} ';')
+  echo "${#FOLDERS[@]} folders to upload found"
 
   # upload the RIF (.csv) files to S3 ETL bucket(s); once per environment, passed in as arg
   source .venv/bin/activate
   for s3_bucket in "${S3_BUCKETS[@]}"; do
     echo "uploading RIF files to: ${s3_bucket}"
     python3 ./s3_utilities.py "${BFD_SYNTHEA_OUTPUT_LOCATION}" "upload_synthea_results" "${s3_bucket}"
+    
+    if [ "$GENERATE_FUTURE" ]; then
```
Comment (mjburling) [inline ops/ccs-ops-misc/synthetic-data/scripts/synthea-automation/build_run_synthea.sh:362]: This is **always** _truthy_ with `GENERATE_FUTURE` defaulting to the string "false", e.g. `[ "false" ] && echo hi mom` and `[ "true" ] && echo hi mom` yield the same result. Borrowing from a similar conditional just above solves for this:
```suggestion
    if [ "$GENERATE_FUTURE" == 'true' ]; then
```

<details><summary>alternatively...</summary> 

You _could_ default `GENERATE_FUTURE` to an empty string, as `[ "" ]` is _falsey_, but I think the above is fine.

</details>

**Review by mjburling — CHANGES_REQUESTED** [review]
👋 Hey, @lsmitchell–apologies for the delay! This is all seems rather practical and in-line with what we've discussed.

We've talked about this in the recent past, so I'm continuing to avoid comments on style until we have more formally adopted some stylistic standards, **however** I have left one comment on the `GENERATE_FUTURE` conditional in the `build_run_sythea.sh` script. Aside from that minor detail, this looks like it's ready to go!

File: ops/ccs-ops-misc/synthetic-data/scripts/synthea-automation/build_run_synthea.sh (hunk 1 above)
Comment (lsmitchell) [reply ops/ccs-ops-misc/synthetic-data/scripts/synthea-automation/build_run_synthea.sh]: Thanks, good catch! I guess i didn't catch this cause in the case there were no folders generated, it just did nothing anyway, but good to have it fixed to be correct.

**Review by mjburling — APPROVED** [review]

**Review by mackec1 — APPROVED** [review]

**Pull request merged**